File: minDepth/mindepth.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

# [3,9,20,null,null,15,7]
class Solution:
    def minDepth(self, root) -> int:
        def recursion(node):
            if  not node:
                return 0
            elif node.left == 0:
                return 1 + recursion(node.right)
            else:
                return 1 + recursion(node.left)
        return recursion(root)

Tree = TreeNode(20,15,7)
Tre = TreeNode(3,9,Tree)

Sol =Solution()

print(Sol.minDepth(Tre))

File: minDepth/mindepth.cpp
#include <iostream>
#include <vector>

using namespace std;

//   Definition for a binary tree node.
struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
    int minDepth(TreeNode *root)
    {
        return 0;
    }
};

int main()
{
    TreeNode Left(5);
    TreeNode Right(6)
        TreeNode Final(6, Left, Right);
    return 0;
}